pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -22,6 +25,14 @@ type LifecycleProcessor struct {
 	DetectLambdaLibrary func() bool
 }
 
+var traceEnabled, _ = strconv.ParseBool(os.Getenv("DD_TRACE_ENABLED"))
+var managedServiceEnabled, _ = strconv.ParseBool(os.Getenv("DD_TRACE_MANAGED_SERVICES"))
+
+// InferredSpansEnabled tells us if the Env Vars are enabled
+// for inferred spans to be created
+var InferredSpansEnabled = traceEnabled && managedServiceEnabled
+var inferredSpan inferredspan.InferredSpan
```

Comment:
I think those 4 lines should be in a function, otherwise it might be hard to test

Reply:
I can put the bool logic into a function for sure, but the `inferredSpan` is needed in both `OnInvokeStart` and `OnInvokeEnd`. This is why I declare it but only initialize it once an invocation starts. Were you suggesting that we declare it inside an initialization function that also handles the bool logic?